code/modules/surgery/mcomp_tendwounds.dm

```diff
@@ -0,0 +1,171 @@
+		return FALSE
+	return TRUE
+
+/datum/surgery_step/mtend_wounds/preop(mob/user, mob/living/carbon/human/target, target_zone, obj/item/tool, tool_type, datum/surgery/surgery)
+	if(user == target)
+		user.visible_message(SPAN_NOTICE("[user] begins to treat the stabilized wounds on their  body with \the [tool]."),
+		SPAN_HELPFUL("You <b>begin to treat your stabilized wounds with \the [tool]"))
+		playsound(target,'sound/misc/heal_gun.ogg',25)
+	else
+		user.affected_message(target,
+			SPAN_HELPFUL("You begin to treat the stabilized wounds on [target]'s body with \the [tool]."),
+			SPAN_HELPFUL("[user] begins to treat your stabilized wounds on your body with \the [tool]."),
+			SPAN_NOTICE("[user] begins to treat the stabilized wounds on [target]'s body with \the [tool]."))
+		playsound(target,'sound/misc/heal_gun.ogg',25)
+
+	target.custom_pain("It feels like your body is being stabbed with needles - because it is!")
+	log_interact(user, target, "[key_name(user)] began tending wounds on [key_name(target)] with \the [tool], starting [surgery].")
+
+/datum/surgery_step/mtend_wounds/success(mob/user, mob/living/carbon/human/target, target_zone, obj/item/tool, tool_type, datum/surgery/surgery)
+	target.heal_overall_damage(60,60)
+
+	if(isSpeciesYautja(target))
+		target.emote("click")
+	else
+		target.emote("pain")
+
+	if(!target.getBruteLoss() && !target.getFireLoss())
+		if(user == target)
+			user.visible_message(SPAN_NOTICE("[user] finshes treating the stabilized wounds on their  body with \the [tool]."),
+			SPAN_HELPFUL("You finish treating the stabilized wounds on your body with \the [tool]"))
+	else
+		user.affected_message(target,
+			SPAN_HELPFUL("You finish treating [target]'s stabilized wounds with \the [tool]."),
+			SPAN_HELPFUL("[user] finished treating your stabilized wounds with \the [tool]."),
+			SPAN_NOTICE("[user] finished treating [target]'s stabilized wounds with \the [tool]."))
+
+	if(!istype(tool, /obj/item/tool/surgery/healing_gun))
+		return
+	var/obj/item/tool/surgery/healing_gun/gun = tool
+	gun.loaded = FALSE
+
+/datum/surgery_step/mtend_wounds/failure(mob/user, mob/living/carbon/human/target, target_zone, obj/item/tool, tool_type, datum/surgery/surgery)
+	log_interact(user, target, "[key_name(user)] failed to tend [key_name(target)]'s wounds with \the [tool], possibly ending [surgery].")
+	return FALSE
+
+/datum/surgery_step/cauterize/mclamp_wound
+	name = "clamp wounds"
+	desc = "clamp the wounds"
+	tools = SURGERY_TOOLS_MEDICOMP_CLAMP_WOUND
+	time = 10 SECONDS
+
+/datum/surgery_step/cauterize/mclamp_wound/tool_check(mob/user, obj/item/tool, datum/surgery/surgery)
+	. = ..()
+	if((. in tools_lit) && !tool.heat_source)
+		return FALSE
+
+/datum/surgery_step/cauterize/mclamp_wound/preop(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, tool_type, datum/surgery/surgery)
+	if(user == target)
+		user.visible_message(SPAN_NOTICE("[user] begins to clamp the treated wounds on their body with \the [tool] ."),
```

Comment:
it isn't clamping if they're not using the clamp, like a lit welding torch or whatever